client/src/components/Layout.tsx

```diff
@@ -29,6 +32,11 @@ function Layout(ownProps: Props) {
     showToast(props.web3Initialized);
   }, [props.web3Initialized]);
 
+  // check the network and show a modal if not on mainnet optimism
+  if (props.web3Error !== undefined && chain?.id !== props.chainID) {
```

Comment:
@codenamejason  if web3Error is set, props.chainID is undefined so we could even skip the second check.
One thing we can do is having `state.web.error` to be a typed error so that we can understand which type of error it is without comparing string. one way could be:

```
// in actions/web3
const Web3BadChainError = "WEB3_BAD_CHAIN_ERROR";

export type Web3Errors = typeof Web3BadChainError | string;

// this would be the state in reducers/web3, where error is currently string | undefined
interface Web3State {
    error: Web3Errors | undefined;
}

// just a simple function to simulate the checks of the error

function Layout(state: Web3State) {
    if (state.error === Web3BadChainError) {
        console.log("bad chain");
        return;
    }

    if (state.error !== undefined) {
        console.log("generic error", state.error);
        return;
    }

    console.log("no errors");
}

Layout({error: Web3BadChainError});
Layout({error: "wallet not connected"});
Layout({error: undefined});
```